src/element.ts

```diff
@@ -9,22 +9,42 @@
 import { Vex } from './vex';
 import { Registry } from './registry';
 import { Flow } from './tables';
+import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext, ElementStyle, ElementAttributes } from './types/common';
 
-export class Element {
-  static newID() {
-    return 'auto' + Element.ID++;
+export abstract class Element {
+  protected static ID: number = 1000;
+
+  protected context?: RenderContext;
+
+  protected rendered: boolean;
+
+  protected style?: ElementStyle;
+
+  protected attrs: ElementAttributes;
+
+  protected classes: Record<string, boolean>;
+
+  protected boundingBox?: BoundingBox;
+
+  protected fontStack: Font[];
+
+  protected musicFont: Font;
+
+  protected registry?: Registry;
+
+  static newID(): string {
+    return `auto${Element.ID++}`;
   }
 
-  constructor({ type } = {}) {
```

Comment:
@0xfe I removed the parameters because they are not used on any of the references. It is also misleading to accept an `ElementAttributes` and ignore all the other fields. Is that fine?

Reply:
The downstream effect of this is not clear to me -- there are other users that extend `Element`, so it could be called with `super({type: "FooNote"})`. I'd prefer to keep this (as a string.)

Reply:
Ok I change to string then.